stablehlo/reference/Tensor.cpp

```diff
@@ -339,52 +331,130 @@ void Tensor::dump() const {
   llvm::errs() << "\n";
 }
 
-Tensor makeTensor(ShapedType type, ArrayRef<StringRef> strData) {
+Tensor makeTensor(DenseElementsAttr attr) {
+  auto type = attr.getType();
   auto elemType = type.getElementType();
 
-  // We are not using parseAttribute for parsing Float literals mainly because
-  // it does not parse special float values like nan, +/-inf.
-  if (auto complexTy = elemType.dyn_cast<ComplexType>()) {
-    auto complexElemTy = complexTy.getElementType();
-    auto floatType = complexElemTy.dyn_cast<FloatType>();
-    if (!floatType) {
-      auto err = invalidArgument("Unsupported element type %s for complex type",
-                                 debugString(complexElemTy).c_str());
-      report_fatal_error(std::move(err));
-    }
-
-    auto floatValues = llvm::to_vector(
-        llvm::map_range(strData, [&](StringRef strNum) -> APFloat {
-          return APFloat(floatType.getFloatSemantics(), strNum);
+  // Handle floating-point types.
```

Comment:
Let's leave a comment about why we are using `uint16_t` to store `f16` and `bf16`.